modules/metastore/src/Controller/MetastoreController.php

```diff
@@ -122,9 +122,16 @@ public function getAll(string $schema_id, Request $request) {
    *
    * @return \Symfony\Component\HttpFoundation\JsonResponse
    *   The json response.
+   *
+   * @throws \InvalidArgumentException
+   *   When an unpublished or invalid resource is requested.
    */
   public function get(string $schema_id, string $identifier, Request $request) {
     try {
+      if (!$this->service->isPublished($schema_id, $identifier)) {
+        throw new \InvalidArgumentException("Error retrieving published dataset: {$schema_id} {$identifier} not found.");
+      }
```

Comment:
Not clear on why this is necessary; the [service get method](https://github.com/GetDKAN/dkan/blob/387959f740b9734fcc01808d612ba16b5591042a/modules/metastore/src/Service.php#L241) will only [retrieve published datasets](https://github.com/GetDKAN/dkan/blob/387959f740b9734fcc01808d612ba16b5591042a/modules/metastore/src/Storage/Data.php#L187). 

If it is necessary, I don't think `InvalidArgumentException` is appropriate here. The arguments are valid. You could use `Drupal\metastore\Exception\MissingObjectException`.

Reply:
I think you're right. This was an early change that was made which can be removed due to later refactoring.